2007-09-04  Guillaume Knispel <gknispel@proformatique.com>

	pf_h323_fail_load_no_crash.patch
	$Revision$
	$Date$
	Bugfix

	When an error is detected during the chan_h323 load phase, most of the
	time the H323 management thread is not yet launched, but then chan_h323
	unload_module() is called just after and tries to manipulate the
	missing thread, eventually leading to a complete Asterisk crash.

	This patch fixes that.

Index: asterisk-1.2.27/channels/chan_h323.c
===================================================================
--- asterisk-1.2.27.orig/channels/chan_h323.c	2008-03-19 11:34:17.000000000 +0100
+++ asterisk-1.2.27/channels/chan_h323.c	2008-03-19 11:34:20.000000000 +0100
@@ -2460,8 +2460,8 @@
 		return -1;
 	}
 	if (!ast_mutex_lock(&monlock)) {
-		if (monitor_thread && (monitor_thread != AST_PTHREADT_STOP)) {
-			/* this causes a seg, anyone know why? */
+		if (monitor_thread && (monitor_thread != AST_PTHREADT_NULL)
+		    && (monitor_thread != AST_PTHREADT_STOP)) {
 			pthread_cancel(monitor_thread);
 			pthread_kill(monitor_thread, SIGURG);
 			pthread_join(monitor_thread, NULL);
